lib/iris/cube.py

```diff
@@ -880,8 +880,11 @@ def convert_units(self, unit):
 
         """
         # If the cube has units convert the data.
-        if not self.units.is_unknown():
-            self.data = self.units.convert(self.data, unit)
+        if self.units.is_unknown():
+            raise iris.exceptions.UnitConversionError(
+                'cannot convert from unknown units to {!s}, '
+                'set the "units" attribute instead'.format(unit))
```

Comment:
> I disagree

Great :+1: I had exactly the same thought process but went for suggesting the easy solution, which was just to remove that section. What I think might confuse is the current wording which I don't think makes clear quite what needs to be done. I'm also not sure how common it is for other error messages in Iris to give a recommendation on how to fix the problem, which was my other, uncommented concern here.

That said I can certainly see the benefit of giving a suggestion in this case. But to be clear that might have to look like the following...
```python
("cannot convert from 'unknown' units to {unit!r}. "
"Consider setting the cube's units instead, with `cube.units = cf_units.Unit({unit!r})`.".format(unit=unit.origin))
```

And maybe at this point we return to the second line confusing rather than helping...